lib/src/insertion_order_parser.dart

```diff
@@ -0,0 +1,199 @@
+      ..insertionOrderId = map['insertionOrderId'] ?? _emptyEntry
+      ..displayName = map['displayName'] ?? _emptyEntry
+      ..updateTime = map['updateTime'] ?? _emptyEntry
+      ..entityStatus = _createEntityStatus(map['entityStatus'])
+      ..pacing = _createPacing(map['pacing'])
+      ..budget = _createBudget(map['budget']);
+
+    return insertionOrder;
+  }
+
+  /// Creates an [InsertionOrder_Pacing] instance from [map].
+  static InsertionOrder_Pacing _createPacing(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) return InsertionOrder_Pacing();
+
+    final pacing = InsertionOrder_Pacing()
+      ..pacingPeriod = _createPacingPeriod(map['pacingPeriod'])
+      ..pacingType = _createPacingType(map['pacingType'])
+      ..dailyMaxMicros = map['dailyMaxMicros'] ?? _emptyEntry
+      ..dailyMaxImpressions = map['dailyMaxImpressions'] ?? _emptyEntry;
+
+    return pacing;
+  }
+
+  /// Creates an [InsertionOrder_InsertionOrderBudget] instance from [map].
+  static InsertionOrder_InsertionOrderBudget _createBudget(
+      Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget();
+    }
+
+    final budget = InsertionOrder_InsertionOrderBudget()
+      ..budgetUnit = _createBudgetUnit(map['budgetUnit'])
+      ..automationType = _createAutomationType(map['automationType']);
+
+    for (Map segmentMap in map['budgetSegments'] ?? []) {
+      budget.budgetSegments.add(_createBudgetSegment(segmentMap));
+    }
+    return budget;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment]
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment
+      _createBudgetSegment(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment();
+    }
+
+    final budgetSegment =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment()
+          ..budgetAmountMicros = map['budgetAmountMicros'] ?? _emptyEntry
+          ..description = map['description'] ?? _emptyEntry
+          ..campaignBudgetId = map['campaignBudgetId'] ?? _emptyEntry
+          ..dateRange = _createDateRange(map['dateRange']);
+
+    return budgetSegment;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange]
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      _createDateRange(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange();
+    }
+
+    final dateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+          ..startDate = _createDate(map['startDate'])
+          ..endDate = _createDate(map['endDate']);
+
+    return dateRange;
+  }
+
+  /// Creates an
```

Comment:
opt: documenting private methods is optional so you can remove it if you think the method name is descriptive enough